### ISSUE TITLE
istio-pilot raise "Omitting listener for management address", and service raise 503 

### ISSUE THREAD

**Issue by johnzheng1975** [istio/istio#9758]
**Describe the bug**
1  Sometimes it will raise 503 or 0 (timeout), when pod liveness access .../v1/health 
![image](https://user-images.githubusercontent.com/17720035/48045345-f384a500-e1ca-11e8-9882-4959a5a377fa.png)

2  Lots of warning in istio-pilot log
```
2018-11-06T03:20:25.791372Z     warn    Omitting listener for management address 10.219.74.100_3000 ({%!s(*core.Address_SocketAddress=&{0xc4330f34c0})}) due to collision with service listener 10.219.74.100_3000 ({%!s(*core.Address_SocketAddress=&{0xc435ccd400})})
2018-11-06T03:20:25.842932Z     warn    Omitting listener for management address 10.219.78.214_3000 ({%!s(*core.Address_SocketAddress=&{0xc431235b40})}) due to collision with service listener 10.219.78.214_3000 ({%!s(*core.Address_SocketAddress=&{0xc430230280})})
2018-11-06T03:20:25.860165Z     warn    Omitting listener for management address 10.219.66.26_8081 ({%!s(*core.Address_SocketAddress=&{0xc429abf1c0})}) due to collision with service listener 10.219.66.26_8081 ({%!s(*core.Address_SocketAddress=&{0xc440ad22c0})})
2018-11-06T03:20:25.888179Z     warn    Omitting listener for management address 10.219.75.204_9090 ({%!s(*core.Address_SocketAddress=&{0xc43873a380})}) due to collision with service listener 10.219.75.204_9090 ({%!s(*core.Address_SocketAddress=&{0xc434cf6280})})
2018-11-06T03:20:25.911444Z     warn    Omitting listener for management address 10.219.67.100_3000 ({%!s(*core.Address_SocketAddress=&{0xc423282940})}) due to collision with service listener 10.219.67.100_3000 ({%!s(*core.Address_SocketAddress=&{0xc42d83aac0})})
2018-11-06T03:20:25.929251Z     warn    Omitting listener for management address 10.219.70.165_8080 ({%!s(*core.Address_SocketAddress=&{0xc42feb3f40})}) due to collision with service listener 10.219.70.165_8080 ({%!s(*core.Address_SocketAddress=&{0xc4243f0640})})
2018-11-06T03:20:25.938594Z     warn    Omitting listener for management address 10.219.66.127_8081 ({%!s(*core.Address_SocketAddress=&{0xc43c483600})}) due to collision with service listener 10.219.66.127_8081 ({%!s(*core.Address_SocketAddress=&{0xc4236d3f40})})
2018-11-06T03:20:25.951273Z     warn    Omitting listener for management address 10.219.66.27_8080 ({%!s(*core.Address_SocketAddress=&{0xc42bd50e00})}) due to collision with service listener 10.219.66.27_8080 ({%!s(*core.Address_SocketAddress=&{0xc434770140})})
2018-11-06T03:20:25.994624Z     warn    Omitting listener for management address 10.219.70.49_3000 ({%!s(*core.Address_SocketAddress=&{0xc42c6a6900})}) due to collision with service listener 10.219.70.49_3000 ({%!s(*core.Address_SocketAddress=&{0xc420de6f00})})
2018-11-06T03:20:26.033900Z     warn    Omitting listener for management address 10.219.78.125_3000 ({%!s(*core.Address_SocketAddress=&{0xc43aa08ac0})}) due to collision with service listener 10.219.78.125_3000 ({%!s(*core.Address_SocketAddress=&{0xc43a58d680})})
2018-11-06T03:20:26.043248Z     warn    Omitting listener for management address 10.219.67.205_3000 ({%!s(*core.Address_SocketAddress=&{0xc424532dc0})}) due to collision with service listener 10.219.67.205_3000 ({%!s(*core.Address_SocketAddress=&{0xc43a58dc80})})
2018-11-06T03:20:26.066873Z     warn    Omitting listener for management address 10.219.72.83_3000 ({%!s(*core.Address_SocketAddress=&{0xc420a8a5c0})}) due to collision with service listener 10.219.72.83_3000 ({%!s(*core.Address_SocketAddress=&{0xc4356c2d80})})
2018-11-06T03:20:26.094076Z     warn    Omitting listener for management address 10.219.74.120_3000 ({%!s(*core.Address_SocketAddress=&{0xc424dcaa80})}) due to collision with service listener 10.219.74.120_3000 ({%!s(*core.Address_SocketAddress=&{0xc420d242c0})})
2018-11-06T03:20:26.115554Z     warn    Omitting listener for management address 10.219.75.140_8080 ({%!s(*core.Address_SocketAddress=&{0xc420f61e40})}) due to collision with service listener 10.219.75.140_8080 ({%!s(*core.Address_SocketAddress=&{0xc42a8666c0})})
2018-11-06T03:20:26.125138Z     warn    Omitting listener for management address 10.219.78.174_9090 ({%!s(*core.Address_SocketAddress=&{0xc42501d040})}) due to collision with service listener 10.219.78.174_9090 ({%!s(*core.Address_SocketAddress=&{0xc434770180})})
2018-11-06T03:20:26.136788Z     warn    Omitting listener for management address 10.219.72.113_3000 ({%!s(*core.Address_SocketAddress=&{0xc43b181500})}) due to collision with service listener 10.219.72.113_3000 ({%!s(*core.Address_SocketAddress=&{0xc442509d40})})
2018-11-06T03:20:26.143513Z     warn    Omitting listener for management address 10.219.72.218_3000 ({%!s(*core.Address_SocketAddress=&{0xc43d5fe800})}) due to collision with service listener 10.219.72.218_3000 ({%!s(*core.Address_SocketAddress=&{0xc4349cd580})})
2018-11-06T03:20:26.174483Z     warn    Omitting listener for management address 10.219.75.217_3000 ({%!s(*core.Address_SocketAddress=&{0xc42422fe00})}) due to collision with service listener 
```
 
3  isto-pilot container cpu usage arrive 2 CPU.
![image](https://user-images.githubusercontent.com/17720035/48045803-b7524400-e1cc-11e8-9fb1-8276e21586de.png)

**Expected behavior**
liveness will always work.

**Steps to reproduce the bug**
Restart all pods in k8s environment, this issue will disappear.
After several days, this issue will raise again

**Version**
kubernetes 1.9.5/ 1.12
istio 1.0/ 1.1, not auth enabled.
cilium 1.2.2/ 1.4

**Installation**
helm

**Environment**
CoreOS 1800.7.0

**Cluster state**
 Product enviroment.

**Comment by Stono** [comment]
This is your liveness probes, I think - can you paste your liveness and readiness probes here?

**Comment by johnzheng1975** [comment]
@Stono , I do not think it is liveness problems.
In most of time, liveness works well.
However, if warning raise in istio-pilot/ istio-pilot load is high, liveness may not work  easily.

**Comment by johnzheng1975** [comment]
Since I am OOO, I will paste my liveness when I am back...

**Comment by mehemken** [comment]
I am also seeing these warnings. It is blocking my deployments. Istio behavior is unpredictable...

**Comment by dkushner** [comment]
We're seeing this behaviour as well in a multi-network Istio deployment. The remote clusters have joined the mesh but pilot appears to be frequently pushing ADS updates with a number of sidecars listed under `pilot_no_ip` category. The actual sidecars that appear in this category vary with each ADS update. Can anyone provide an explanation as to whats going on here? Nothing else from the control plane logs indicates any issue that I can find.

**Comment by johnzheng1975** [comment]
Here is our probe, FYI. Is there anything wrong? Thanks.
```
  livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /backoffice/v1/health
            port: 9090
            scheme: HTTP
          initialDelaySeconds: 360
          periodSeconds: 20
          successThreshold: 1
          timeoutSeconds: 5
        name: backoffice
        ports:
        - containerPort: 9090
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /backoffice/v1/health
            port: 9090
            scheme: HTTP
          initialDelaySeconds: 60
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 30
```

**Comment by howardjohn** [comment]
Does this still happen on new Istio versions? Istio 1.1 is EOL (https://istio.io/blog/2019/announcing-1.1-eol/), we made some fixes to this part of the code in 1.3 -- you should never see "omitting listener for management address"
